src/scenes/LanguageSelection.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component, PropTypes } from 'react';
+import { AsyncStorage, Image, StyleSheet, View } from 'react-native';
+
+import Spinner from 'react-native-loading-spinner-overlay';
+import { connect } from 'react-redux';
+import { Drawer, Button } from 'react-native-material-design';
+
+import DrawerCommons from '../utils/DrawerCommons';
+import I18n from '../constants/Messages';
+
+
+class LanguageSelection extends Component {
+
+    static contextTypes = {
+        drawer: PropTypes.object.isRequired,
+        navigator: PropTypes.object.isRequired
+    };
+
+    static renderLoadingView() {
+        return (
+            <View style={{ flex: 1 }}>
+                <Spinner
+                    overlayColor="#EEE"
+                    visible
+                />
+            </View>
+        );
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            loaded: false
+        };
+        this.drawerCommons = new DrawerCommons(this);
+    }
+
+    componentDidMount() {
+        this.checkLanguageSelected().done();
+    }
+
+    async checkLanguageSelected() {
+        const isLanguageSelected = await AsyncStorage.getItem('isLanguageSelected');
+        const code = await AsyncStorage.getItem('langCode');
+        const theme = await AsyncStorage.getItem('theme');
+        const color = await AsyncStorage.getItem('color');
+        this.setState({'loaded': true});
+        if (JSON.parse(isLanguageSelected)) {
+            if (code) {
+                this.drawerCommons.changeLanguage(code, false);
+            }
+            if (theme && color) {
+                this.drawerCommons.changeTheme(theme, color, false);
+            }
+            this.context.navigator.to('countryChoice');
+        }
+    }
+
+    onPress() {
+        AsyncStorage.setItem('isLanguageSelected', JSON.stringify(true));
+        this.context.navigator.to('countryChoice');
+    }
+
+    render() {
+        if (!this.state.loaded) {
+            return LanguageSelection.renderLoadingView();
```

Comment:
I added component for loader in that PR: https://github.com/theirc/app.refugee.info/pull/71/files#diff-5a3a1871501fb420969ab552c6b7a429R88